### DISCUSSION TITLE
How to enable dark theme?

### DISCUSSION THREAD

**Discussion by Sbrjt** [pawelsalawa/sqlitestudio#4992] [Q&A] [answered]
I don't want to install any other additional software for this. Is it possible to so natively on Windows?

**Comment by pawelsalawa** [comment, marked as answer]
No need to install anything extra. Just go to Configuration dialog -> Look & Feel -> Select "Fusion Dark" theme.

**Reply by Sbrjt** [reply]
I think you should make it the default bcoz most of the users are developers and developers like dark theme ;)

**Comment by Sbrjt** [comment]
So go to Tools > Open Configuration dialog > Look & Feel > Style > Set Current style: Fusion Dark > OK 🎉

**Comment by audiomuze** [comment]
What does one need to do t have Fusion Dark appear in the dropdown list?

I have 31 code windows in SQLitestudio, which are all stored in unintelligible form in /home/username/.config/SalSoft/SQLiteStudio/settings3.   If I delete the settings3 file (and lose all my code windows and snippets) Sqlitestudio fires up in dark mode.  It feels to me like there's  a bug in color theme support - there's no way to get any current config reset that reverts to following the desktop theme, without losing all prior work or spending time copying all code windows and code snippets into a different environment so they can be copied back again.  Why can't the snippets and code windows live as a SQLite table rather than be obfuscated and tied to some binary configuration file format?

Here's what the defaults look like when running on Breeze Dark in KDE Plasma:

![image](https://github.com/user-attachments/assets/8dbb6b17-9d62-421b-89b8-cfaab58eb650)

There a way to get rid of that horrible, eye burning white background and set it to something less offensive?  And why is the coding window background so dark with no way to change the background in any of the windows?

The light theme is very hard on light sensitive eyes and the dark theme is great for browsing but terrible for coding.

I don't think it's a theme or Qt thing - I'm running many apps that leverage Qt and SqliteStudio is the only app with these colour issues.

**Reply by pawelsalawa** [reply]
What about this reset button?
![image](https://github.com/user-attachments/assets/9104a811-10c3-45a6-aa71-2629595e4cf2)

**Reply by audiomuze** [reply]
Tried it, to no effect

**Reply by pawelsalawa** [reply]
The `/home/username/.config/SalSoft/SQLiteStudio/settings3` file is indeed SQLite3 database, although some internal values (such as SQL Editor contents) are serialized QHash objects (binary, as you noticed) and then this serialized form is written to a BLOB column of settings. It is this way simply because it's faster (not faster to write such code, but it's faster to save/load).

Let me clarify something. When you change theme in SQLiteStudio to Fusion Dark and then you shut down SQLiteStudio and then run it again - is the theme still wrong?

**Reply by audiomuze** [reply]
> Let me clarify something. When you change theme in SQLiteStudio to Fusion Dark and then you shut down SQLiteStudio and then run it again - is the theme still wrong?

I'm running KDE Plasma on Arch Linux.  My default global theme is Breeze Dark.  There is no Fusion or Fusion Dark installed in my desktop environment
![image](https://github.com/user-attachments/assets/7e965d6c-f9fe-45d2-87e7-4a6d2208f162)

If I delete  ```/home/username/.config/SalSoft/SQLiteStudio/settings3``` then SQLiteStudio fires up looking like this:
![image](https://github.com/user-attachments/assets/cf554694-757a-49dd-a016-cd1de65dd2ff)

If I then shut down and restart SQLiteStudio it fires up looking like this:
![image](https://github.com/user-attachments/assets/064cc710-5a30-4e7b-a691-cb307fe4db38)


Under Look & feel/Syle  the only dropdown items at Fusion and Windows.

**Reply by pawelsalawa** [reply]
I guess you are using SQLiteStudio from Arch repo or you compiled it by yourself? Binaries from sqlitestudio.pl come with Fusion Dark included.

If you use generic `Fusion` theme and switch your KDE theme, I suppose KDE is applying CSS stylesheets to all running Qt apps - that's why it works instantly after you change KDE theme, but then if you restart SQLiteStudio, this is no longer correctly handled.

Frankly speaking I don't know if KDE or SQLiteStudio is to blame here. I will require closer look from me or one of contributors. Fortunately now it's clear what the problem is and I will link this info in the issue report.

In the meantime, perhaps it's worth to try binaries from the homepage https://sqlitestudio.pl ?

**Reply by audiomuze** [reply]
I was running from the Arch repo, I've uninstalled that and installed the binaries from the homepage as you suggested, reinstatated my config file, selected Fusion Dark and reset code syntax colours to defaults.  That seems to have done the trick.  Thank you.
